src/test/libp2p/ping/test_ping.rs

```diff
@@ -0,0 +1,47 @@
+use futures::prelude::*;
+use libp2p::{multiaddr::Protocol, noise, ping, swarm::SwarmEvent, tcp, yamux, Multiaddr};
+use std::{error::Error, time::Duration};
+
+#[tokio::main]
+async fn main() -> Result<(), Box<dyn Error>> {
+    // Usage: ./test_libp2p_ping [listen-port] [peer-multiaddr]
+
+    let mut swarm = libp2p::SwarmBuilder::with_new_identity()
```

Comment:
For those of us who aren't familiar with libp2p, can you add a few more comments about what the program is doing, and what things like `noise::Config::new` and `yamux::Config::default` are used for?